pilot/pkg/config/kube/gateway/templates.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gateway
+
+import (
+	"embed"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"sigs.k8s.io/yaml"
+)
+
+// Templates embeds the templates
+//go:embed templates/*
+var Templates embed.FS
+
+func funcMap() template.FuncMap {
+	f := sprig.TxtFuncMap()
+	f["strdict"] = func(v ...string) map[string]string {
```

Comment:
'strdict' ? At least a comment explaining what it does. 

Does helm have a similar function ? It wouldn't be bad to reuse the names.

Reply:
Yeah the problem is it returns the wrong type unfortunately. added a comment